### PR TITLE
97 post sharing

### PR DESCRIPTION
**Pull request by LordofMankid**
# Description

added sharing and auto closing of modal after sharing.

also added functionality that saves your last known location when prompted to redirect. 
## Issues

unknown edge cases for sharing when users are not logged in

## Test

share a link with yourself or someone with dev build

tap on it

log out, then tap on it
## Possible Downsides

saving last known location might start weird nav bugs esp in development. i have it clear the storage after navigating through the redirect. also sometimes it navigates to home screen first and then to the saved path. 

too lazy to debug that though (need to work on other tickets hehe)

### CODE REVIEW

**Review by johnny-t06 — APPROVED** [review]
LGTM, looks great. Tested before with past dev build

File: src/components/home/OptionModel.tsx [hunk 1]
```diff
@@ -171,11 +168,12 @@ const OptionModel = (props: OptionModelProps) => {
 
                 <Pressable
                   style={styles.optionButton}
-                  onPress={() =>
-                    Sharing.shareAsync(
-                      "https://www.youtube.com/watch?v=5payqF0YypY"
```
Comment (ManuP6789) [inline src/components/home/OptionModel.tsx:176]: Good song 👍 (I thought I was going to get rick rolled)

File: src/components/login/SignupSection.tsx [hunk 2]
```diff
@@ -221,12 +231,19 @@ const SignupSection = () => {
...
             secureTextEntry
-            autoComplete={Platform.OS === "ios" ? "password-new" : "new-password"}
+            autoComplete={
+              Platform.OS === "ios" ? "password-new" : "new-password"
+            }
             onChangeText={(text) => {
               validateNewPassword(text);
             }}
           />
-          <Text style={styles.errorText}>{errors.newError}</Text>
```
Comment (ManuP6789) [inline src/components/login/SignupSection.tsx]: There is an issue here with the spacing on the signup section, If you conditionally render the error, it will move whatever is below the error.

**Review by ManuP6789 — APPROVED** [review]
LGTM, added comment on error message in signup page should be easy fix!
